turbinia/jobs/analysis/http_access_logs.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Job to execute HTTP Access logs analysis task."""
+from __future__ import unicode_literals
+
+from turbinia.workers import artifact
+
+from turbinia.evidence import Directory
+from turbinia.evidence import RawDisk
+from turbinia.evidence import GoogleCloudDisk
+from turbinia.evidence import GoogleCloudDiskRawEmbedded
+from turbinia.evidence import ExportedFileArtifact
+from turbinia.evidence import ReportText
+
+from turbinia.jobs import TurbiniaJob
+from turbinia.workers.analysis import wordpress
+
+ACCESS_LOG_ARTIFACTS = [
+    'DockerContainerLogs',
+    'NginxAccessLogs',
+    'ApacheAccessLogs'
+]
+
+class HTTPAccessLogExtractionJob(TurbiniaJob):
+  """HTTP Access log extraction job."""
+
+  evidence_input = [
+      Directory, RawDisk, GoogleCloudDisk, GoogleCloudDiskRawEmbedded]
+
+  evidence_output = [ExportedFileArtifact]
+
+  def __init__(self):
+    super(HTTPAccessLogExtractionJob, self).__init__(
+        name='HTTPAccessLogExtractionJob')
+
+  def create_tasks(self, evidence):
+    """Create task.
+
+    Args:
+      evidence: List of evidence object to process
+
+    Returns:
+        A list of tasks to schedule.
+    """
+    tasks = []
+    for artifact_name in ACCESS_LOG_ARTIFACTS:
+      tasks.extend([artifact.FileArtifactExtractionTask(artifact_name) for _
+                    in evidence])
+    return tasks
+
+class HTTPAccessLogAnalysisJob(TurbiniaJob):
+  """HTTP Access log analysis job."""
+   # Types of evidence that this Job will process.
+  evidence_input = [ExportedFileArtifact]
+  evidence_output = [ReportText]
+
+  def __init__(self):
+    super(HTTPAccessLogAnalysisJob, self).__init__(
+        name='HTTPAccessLogAnalysisJob')
+
+  def create_tasks(self, evidence):
+    """Create task.
+    Args:
+      evidence: List of evidence object to process
+    Returns:
+        A list of tasks to schedule.
+    """
+    tasks = []
```

Comment:
Could also just say:
return [e for e in evidence if e.artifact_name in ACCESS_LOG_ARTIFACTS]

Reply:
Uuh, wouldn't that would return a list of `evidence` objects instead of a list of tasks? I thought we wanted the latter.

What my code currently does is the equilvalent of 

`[wordpress.WordPressAccessLogAnalysisTask() for e in evidence if e.artifact_name in ACCESS_LOG_ARTIFACTS]`, which IMO is a bit too long :)

I could change it to

tasks = [e for e in evidence if e.artifact_name in ACCESS_LOG_ARTIFACTS]
return [wordpress.WordPressAccessLogAnalysisTask() for _ in tasks]